src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/azuredevops/AzureDevOpsServerPullRequestDecorator.java

```diff
@@ -0,0 +1,407 @@
+
+    private String getStatusApiUrl(String azureUrl, String azureProjectId, String azureRepositoryName, String pullRequestId, String apiVersion) {
+        // POST https://{instance}/{project}/_apis/git/repositories/{repositoryId}/pullRequests/{pullRequestId}/statuses?api-version=6.0-preview.1
+        return azureUrl + azureProjectId +
+                "/_apis/git/repositories/" +
+                azureRepositoryName +
+                "/pullRequests/" +
+                pullRequestId +
+                "/statuses" +
+                getApiVersion(apiVersion);
+    }
+
+    private String getThreadApiUrl(String azureUrl, String azureProjectId, String azureRepositoryName, String pullRequestId, String apiVersion) {
+        //POST https://{instance}/{project}/_apis/git/repositories/{repositoryId}/pullRequests/{pullRequestId}/threads?api-version=6.0-preview.1
+        return azureUrl + azureProjectId +
+                "/_apis/git/repositories/" +
+                azureRepositoryName +
+                "/pullRequests/" +
+                pullRequestId +
+                "/threads" +
+                getApiVersion(apiVersion);
+    }
+
+    private String getGitPullRequestStatus(AnalysisDetails analysisDetails) throws IOException {
+        final String gitStatusContextGenre = "SonarQube";
+        final String gitStatusContextName = "QualityGate";
+        final String gitStatusDescription = "SonarQube Gate";
+
+        GitPullRequestStatus status = new GitPullRequestStatus(
+                GitStatusStateMapper.toGitStatusState(analysisDetails.getQualityGateStatus()),
+                gitStatusDescription,
+                new GitStatusContext(gitStatusContextGenre, gitStatusContextName),
+                String.format("%s/dashboard?id=%s&pullRequest=%s", server.getPublicRootUrl(),
+                        URLEncoder.encode(analysisDetails.getAnalysisProjectKey(),
+                                StandardCharsets.UTF_8.name()),
+                        URLEncoder.encode(analysisDetails.getBranchName(),
+                                StandardCharsets.UTF_8.name())
+                )
+        );
+        return new ObjectMapper().writeValueAsString(status);
+    }
+
+    private void sendPost(String apiUrl, String body, String successMessage, String authorizationHeader) throws IOException, MalformedURLException, URISyntaxException {
+        apiUrl = encodeURI(apiUrl);
+        LOGGER.trace(String.format("sendPost: URL: %s ", apiUrl));
+        LOGGER.trace(String.format("sendPost: BODY: %s ", body));
+        HttpPost httpPost = new HttpPost(apiUrl);
+        httpPost.addHeader("Accept", "application/json");
+        httpPost.addHeader("Content-Type", "application/json; charset=utf-8");
+        httpPost.addHeader("Authorization", authorizationHeader);
+        StringEntity entity = new StringEntity(body, StandardCharsets.UTF_8);
+        httpPost.setEntity(entity);
+
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
```

Comment:
Possibly use `createSystem` rather than `createDefault` so that any proxy settings set as System properties are obeyed?

Reply:
Fixed